applications/asset_tracker_v2/src/cloud/cloud_codec/aws_iot_codec.c

```diff
@@ -689,74 +724,191 @@ int cloud_codec_encode_batch_data(
 	cJSON *root_obj = cJSON_CreateObject();
 	cJSON *gps_obj = cJSON_CreateArray();
 	cJSON *sensor_obj = cJSON_CreateArray();
-	cJSON *modem_obj = cJSON_CreateArray();
+	cJSON *modem_dyn_obj = cJSON_CreateArray();
+	cJSON *modem_stat_obj = cJSON_CreateArray();
 	cJSON *ui_obj = cJSON_CreateArray();
 	cJSON *accel_obj = cJSON_CreateArray();
 	cJSON *bat_obj = cJSON_CreateArray();
 
 	if (root_obj == NULL || gps_obj == NULL || sensor_obj == NULL ||
-	    modem_obj == NULL || ui_obj == NULL || accel_obj == NULL ||
-	    bat_obj == NULL) {
+	    modem_dyn_obj == NULL || modem_stat_obj == NULL || ui_obj == NULL ||
+	    accel_obj == NULL || bat_obj == NULL) {
 		cJSON_Delete(root_obj);
 		cJSON_Delete(gps_obj);
 		cJSON_Delete(sensor_obj);
-		cJSON_Delete(modem_obj);
+		cJSON_Delete(modem_stat_obj);
+		cJSON_Delete(modem_dyn_obj);
 		cJSON_Delete(ui_obj);
 		cJSON_Delete(accel_obj);
 		cJSON_Delete(bat_obj);
 		return -ENOMEM;
 	}
 
-	/* GPS data */
-	for (int i = 0; i < gps_buf_count; i++) {
-		if (gps_buf[i].queued) {
-			err += gps_data_add(gps_obj, &gps_buf[i], true);
+	/* UI */
+	while (ring_buf_is_empty(&ui_buf) == 0) {
+		struct cloud_data_ui *ui;
+
+		uint32_t len = ring_buf_get_claim(&ui_buf,
```

Comment:
The docs are a little ambiguous on whether this API can be used on a ring buffer in item mode
```
 * @warning
 * Ring buffer instance should not mix byte access and item access
 * (calls prefixed with ring_buf_item_).
 ````

Reply:
We are using byte mode.

Reply:
Oh, I guess I was confused by the macro defining the buffers: `RING_BUF_ITEM_DECLARE_SIZE`